### DISCUSSION TITLE
how to select rows of budibase table and insert in another table database

### DISCUSSION THREAD

**Discussion by Rudison** [Budibase/budibase#12968] [Help] [answered]
![budibase 1](https://github.com/Budibase/budibase/assets/6667805/86e11aea-1be2-4b77-bea5-1ad9770ed760)

Below I have the sql server database table
![budibase 2](https://github.com/Budibase/budibase/assets/6667805/3260235a-1cdc-4579-985e-3423e3b9666e)

I want to select all records from budibase and insert my table in sql server

**Comment by ConorWebb96** [comment]
Hey @Rudison,

I'd recommend you use an [automation](https://docs.budibase.com/docs/automation-steps) to do this. You can do a range of different things with automations e.g. query tables, then iterate over the table and create rows for a different table using the queried tables data.

Hopefully this helps.

**Comment by Rudison** [comment]
Can you help me @ConorWebb96 

![passo 1](https://github.com/Budibase/budibase/assets/6667805/0883a8ff-a5b1-4fc2-8e53-3f4a495a27c5)

![passo 2 novo](https://github.com/Budibase/budibase/assets/6667805/fe62dfba-32a8-4cd9-b8ce-a9460d3b9cee)

**Reply by ConorWebb96** [reply]
Hey @Rudison,

I'm confused, why do you need an external data connector step? You have everything you need to get the relevant rows from the query step right? You'd then just loop through the data generated from the query step and insert that into your MSSQL table.

If you need to mould or interact with the data in a different way you can use a **JS scripting** block to manipulate it to meet your needs.

**Comment by Rudison** [comment]
The connection with the external database is why I return the last ID of the table, as it is not self-incrementing so I always have to search for the last ID.

**Reply by ConorWebb96** [reply, marked as answer]
Can't you create a query that will select the table to get the last row, this is then used in your external data connector. From here you can use the js scripting block to modify the id in whatever way you wish so that it works with your database?

For mssql if im not mistaken the query would look something like that.

```sql
SELECT TOP 1 *
FROM YourTableName
ORDER BY YourOrderingColumn DESC;
```
